client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -944,13 +921,33 @@ private static void randomizeFetchSourceContextParams(Consumer<FetchSourceContex
         }
     }
 
-    private static void setRandomTimeout(AcknowledgedRequest<?> request, Map<String, String> expectedParams) {
+    private static <T> void setRandomIndicesOptions(Function<IndicesOptions, T> setter, Supplier<IndicesOptions> getter,
+                                                    Map<String, String> expectedParams) {
+
+        if (randomBoolean()) {
+            setter.apply(IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(),
+                randomBoolean()));
+        }
+        expectedParams.put("ignore_unavailable", Boolean.toString(getter.get().ignoreUnavailable()));
+        expectedParams.put("allow_no_indices", Boolean.toString(getter.get().allowNoIndices()));
+        if (getter.get().expandWildcardsOpen() && getter.get().expandWildcardsClosed()) {
+            expectedParams.put("expand_wildcards", "open,closed");
+        } else if (getter.get().expandWildcardsOpen()) {
+            expectedParams.put("expand_wildcards", "open");
+        } else if (getter.get().expandWildcardsClosed()) {
+            expectedParams.put("expand_wildcards", "closed");
+        } else {
+            expectedParams.put("expand_wildcards", "none");
+        }
```

Comment:
I wonder if we should simplify this method. Given that we generate the random indices options as part of the method, we could reuse those rather than using the getter, then the getter argument is not needed. Also you could just return the indices options and that would remove the need for the setter argument too. Thoughts?

Reply:
But if we don't generate an `IndicesOptions`, the `getter` then gets the default for the given request. Without that, we'd have to pass in the default to this method.

So, we could change it to:

```
private static IndicesOptions createAndExpectRandomIndicesOptions(IndicesOptions default, Map<String, String> expectedParams) {
  if (randomBoolean()) {
    ...
  } else {
    //use default
    return null;
  }
```

And then handle the returned value depending on whether it's null or not. Is that nicer, though?

Reply:
good point! I missed the default case. I do not like the `null` invariant that we would introduce here, leave it as-is. Thanks for your patience ;)